sympy/tensor/array/sparse_ndim_array.py

```diff
@@ -109,8 +109,7 @@ def reshape(self, *newshape):
         if new_total_size != self._loop_size:
             raise ValueError("Invalid reshape parameters " + newshape)
 
-        return type(self)(*(newshape + (self._array,)))
-
+        return type(self)(*(newshape + (self._sparse_array,)))
```

Comment:
is this a bugfix?

Reply:
Yes, but apprently the problem isn't resolved. I will discard this change and fix this issue in a seperated PR.